src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_ComboBox.cs

```diff
@@ -931,6 +965,45 @@ protected override void OnCollectionChanged(NotifyCollectionChangedEventArgs e)
 			public TType GetAt(int index) => this[index];
 		}
 
+		public sealed class DisposableDataContext : IDisposable
+		{
+			public DisposableDataContext()
+			{
+				_item = Items[0];
+			}
+
+			public int[] Items { get; set; } = { 0, 1 };
+
+			private int _item;
+			private bool _isDisposed;
+
+			public int Item
+			{
+				get
+				{
+					if (_isDisposed)
+					{
+						Assert.Fail();
+					}
+
+					return _item;
+				}
+				set
+				{
+					if (_isDisposed)
+					{
+						Assert.Fail();
+					}
+
+					_item = value;
```

Comment:
The exception thrown by `Assert.Fail` will get caught here:

https://github.com/unoplatform/uno/blob/99a8c21d1bdae50965420accd2728991eea111e9/src/Uno.UI/DataBinding/BindingPath.cs#L865

Instead of throwing here, introduce a bool property to `DisposableDataContext` and set it to true in this code path, then assert the property value

Reply:
Actually you can just expose `_isDisposed`